src/__tests__/queryByApi.test.js

```diff
@@ -48,7 +48,7 @@ test('queryByText not found', () => {
   ).toBeFalsy();
 });
 
-test('queryByText nested text across multiple <Text> in <Text>', () => {
+test('queryByText does not match nested text across multiple <Text> in <Text>', () => {
```

Comment:
So we're regressing this functionality to be compatible with Testing Library, which also doesn't support this?

Reply:
Yes. I checked, testing library doesn't match in this case.

Reply:
How does this PR change this functionality? Finding the text in nodes still recurses through children

Reply:
Yes https://github.com/callstack/react-native-testing-library/pull/554/files#diff-28f6e2b77eef7ef8b7583339cfdf593fR43

Reply:
I must be missing something but **how** does the logic added to `getNodeByText` end up not matching in this case? As I see it the the logic in `getChildrenAsText ` will return `['Hello', '  ', 'World', '!']` which when joined will be `"Hello World!"`, which is an exact match for what you are looking for.

Reply:
Yes, the change is to keep matching component with multiple children, but not `<Text>Hello <Text>World</Text></Text`, because now `getChildrenAsText` don't read its children's children

Reply:
Ah, so if the test render looked like this:
```javascript
<Text>
  <Text>Hello</Text>
  World!
</Text>
```
Would this then match the outer text?

Reply:
This currently match is the current released version, but wouldn't match in my current implementation. It's the topic of discussion in https://github.com/callstack/react-native-testing-library/pull/554#issuecomment-705437898